render/gles2/texture.c

```diff
@@ -214,7 +214,9 @@ struct wlr_texture *wlr_gles2_texture_from_wl_drm(struct wlr_egl *egl,
 		texture->has_alpha = false;
 		break;
 	case EGL_TEXTURE_RGBA:
+#if defined(EGL_WL_bind_wayland_display) && defined(EGL_TEXTURE_EXTERNAL_WL)
```

Comment:
Since this is now in Khronos' registry, why do we need an `#ifdef` here?

Reply:
It's in Khronos registry, but it's now in all gpu drivers.  Mesa still publish this extension in eglmesaext.h.  It'll be updated at some point.  Each gpu driver updates Khronos headers at different speed. To make wlroots compile on a wider range of drivers (older extension version, older drivers, very recent driver with up-to-date khronos header, etc.), the define protect its usage, only when it's defined.

Reply:
But aren't there a lot more symbols that may not be defined, like `eglBindWaylandDisplayWL`?

I'd prefer not to clutter the code with these `#ifdef`s. If the drivers are outdated, they should get updated. If the drivers are closed-source, we won't do anything to support them.

Reply:
Runtime symbols like eglBindWaylandDisplayWL are normally protected by the dynamic resolution with eglGetProcAddress(), and scanning in EGL_EXTENSIONS string. Compilation time definitions (IDs, typedefs, etc.) can also be protected with ifdefs on the extension (e.g. #ifdef EGL_WL_bind_wayland_display). Here, the issue is that some drivers claim an older version of the extention, with a missing definition.

If wlroots wants to focus only on Mesa (and eventually only Khronos conformant closed-source drivers), then I agree with you this quirk can be removed. I'll update the pull-request by dropping this change, and I'll see if I can fix that issue directly with GPU driver vendors.

Reply:
We want to focus on any open-source Khronos conformant GL implementation. We won't offer support for closed-source drivers because we can't go down the stack and fix issues in the GL implementation in case of a bug. Debugging a black box is counter-productive.